include/deal.II/fe/fe_enriched.h

```diff
@@ -0,0 +1,585 @@
+ * whereas enriched DoFs are coming from a single finite element
+ * <code>FE_Q<dim>(1)</code> used with a single enrichment function
+ * <code>function</code>. In this case, the total number of DoFs on the
+ * enriched element is the sum of DoFs from <code>FE_Q<dim>(2)</code> and
+ * <code>FE_Q<dim>(1)</code>.
+ *
+ * As an example of enrichment function, consider $ \exp(-x) $, which
+ * leads to the following shape functions on unit element:
+ * <table>
+ *   <tr>
+ *     <td align="center">
+ *       @image html fe_enriched_1d.png
+ *     </td>
+ *     <td align="center">
+ *       @image html fe_enriched_h-refinement.png
+ *     </td>
+ *   </tr>
+ *   <tr>
+ *     <td align="center">
+ *       1d element, base and enriched shape functions.
+ *     </td>
+ *     <td align="center">
+ *       enriched shape function corresponding to the central vertex.
+ *     </td>
+ *   </tr>
+ * </table>
+ *
+ * <h3>Using enriched and non-enriched FEs together</h3>
+ *
+ * In most applications it is beneficial to introduce enrichments only in
+ * some part of the domain (e.g. around a crack tip) and use standard FE (e.g. FE_Q)
+ * elsewhere. In order to make the resulting FE space $C^0$ continuous in the
+ * case of continuous enrichmenet functions, all enriched DoFs on the interface
+ * between enriched and non-enriched elements are constrained to be zero.
+ * To get those constraints working automatically, the user needs to wrap a FE
+ * to be used in part of the domain without enrichment:
+ * @code
+ *   FE_Enriched<dim> fe_non_enriched(FE_Q<dim>(1));
+ * @endcode
+ * This constructor is equivalent to calling
+ * @code
+ *   FE_Enriched<dim> fe_non_enriched(FE_Q<dim>(1),
+ *                                    FE_Nothing<dim>(1,true),
+ *                                    NULL);
+ * @endcode
+ * and will result in the correct constraints for enriched DoFs attributed to
+ * support points on the interface between the two regions.
+ *
+ * <h3>Implementation</h3>
+ *
+ * The implementation of the class is based on FESystem which is aggregated as
+ * a private member. The simplest constructor <code> FE_Enriched<dim> fe(FE_Q<dim>(2), FE_Q<dim>(1),function)</code>
+ * will internally initialize FESystem as
+ *
+ * @code
+ * FESystem<dim> fe_system(FE_Q<dim>(2),1,
+ *                         FE_Q<dim>(1),1);
+ * @endcode
+ *
+ * Note that it would not be wise to have this class derived
+ * from FESystem as the latter concatenates the given elements into different
+ * components of a vector element, whereas the current class combines the given
+ * elements into the same components. For instance, if two scalar elements are
+ * given, the resulting element will be scalar rather than have two components
+ * when doing the same with an FESystem.
+ *
+ * The ordering of the shape function, @p interface_constrains, the @p prolongation (embedding)
+ * and the @p restriction matrices are taken from the FESystem class.
+ *
+ * <h3>References</h3>
```

Comment:
I would move that even above the "Implementation" section.

Does the paper actually discuss implementation details, or just show results? In the former case, we may want to consider listing it along with the other papers at the top of `publications.html` that discuss implementations.